lib/aruba/cucumber/command.rb

```diff
@@ -28,6 +28,21 @@
   run_simple(cmd, :fail_on_error => true, :exit_timeout => secs && secs.to_i)
 end
 
+When(/I run the following script:$/) do |file_content|
+  prepend_environment_variable('PATH', expand_path('bin') + ':')
+  step 'an executable named "./bin/myscript" with:', file_content
+  step 'I run `myscript`'
+end
+
+When(/^I run the following (bash|zsh|fish|dash)? ?commands?:$/) do |shell_type, file_content|
```

Comment:
About point 2: 
Yep, I need to be explicity saying what is the shell I'm running command into to test compatibility of commands, cli-under-test with the different shell dialects.
(I'm using aruba to set up acceptance testing for  [resty](https://github.com/micha/resty))